Description: restore UIL callback casts accepted by OpenMotif
 OpenMotif 2.3.8 as used by Debian and Ubuntu does not declare
 uil_message_cb_t or uil_status_cb_t. Using those type names
 prevents dtappbuilder's abuil_load.c from compiling.
 .
 Restore the previous UIL_CB typedef and callback casts while the
 upstream API change is discussed.
Author: Reaper <JohnGrimmReaper@disroot.org>
Origin: revert, upstream commit:1ac240562a0d8c87fa1030e73e27fb40e61ecf9d
Forwarded: no
Last-Update: 2026-08-03

diff --git a/programs/dtappbuilder/src/libABil/abuil_load.c b/programs/dtappbuilder/src/libABil/abuil_load.c
index 1d6c728b5..ee5d3fe67 100644
--- a/programs/dtappbuilder/src/libABil/abuil_load.c
+++ b/programs/dtappbuilder/src/libABil/abuil_load.c
@@ -65,6 +65,9 @@ extern XtPointer _XmStringUngenerate (XmString string,
                         XmTextType output_type);
 
 
+typedef Uil_continue_type(*UIL_CB)();
+
+
 /*
  * Private function declarations
  */
@@ -1137,8 +1140,8 @@ parse_uil(
     /* Call uil compiler with appropriate paramters */
 
     compile_stat = Uil(&command_desc, &compile_desc,
-		       (uil_message_cb_t)message_cb, (char *)&user_mess_data,
-		       (uil_status_cb_t)status_cb, (char *)&user_stat_data);
+		       (UIL_CB)message_cb, (char *)&user_mess_data,
+		       (UIL_CB)status_cb, (char *)&user_stat_data);
 
     fprintf(stderr, "\nUIL Compiler Version %d, ",
 		compile_desc.compiler_version);
